Update Components build and E2E testing instructions

sebastienros opened the pull request:
The steps needed to be updated since the migration to npm.

sebastienros committed:
Update Components build and E2E testing instructions

sebastienros commented on src/Components/README.md:
````diff
@@ -61,10 +73,23 @@ source activate.sh
 
 ```powershell
-dotnet build
+.\startvs.cmd
````
Should it repeat `cd src/Components` since it was done in the previous step?

amcasey replied:
An alternative would be to make both stateless by running, e.g. `src\Components\build.cmd`.

amcasey commented on src/Components/README.md:
````diff
@@ -29,29 +29,41 @@ The following contains a description of each sub-directory in the `Components` d
+You may need to kill some processes holding on files that are being deleted. The following command may help you but be aware that this could also stop other important processes:
+
+```powershell
+Get-Process dotnet, escape-node-job, msbuild, VBCSCompiler, node, vstest.console, Microsoft.CodeAnalysis.LanguageServer -ErrorAction Continue | Stop-Process;
````
I usually use `taskkill`, but this should be similar (and cross-plat).

Shutting down VS before cleaning your repo is usually a good idea.

sebastienros replied:
Thanks, done

amcasey commented on src/Components/README.md:
````diff
@@ -61,10 +73,23 @@ source activate.sh
 ```
 
-4. Now you can build `src\Components\Web.JS\` by running the following commands in the `src\Components\Web.JS` directory:
+1. Now you can build all the JavaScript assets required by the repository (including SignalR for instance) by running the following command:
````
It's probably implied, but it might be helpful to explicitly say that you should run this in the repo root.

sebastienros replied:
I made the commands "stateless" as you suggested. I think it doesn't need to be mentioned anymore as the previous command is also at the root.

sebastienros replied:
... and this command also works in a subfolder apparently (just tried)

amcasey commented on src/Components/README.md:
```diff
@@ -123,4 +136,4 @@ dotnet test --no-build
 ## More Information
 
-For more information, see the [ASP.NET Core README](../../README.md).
+For more information, see the [ASP.NET Core README](https://github.com/dotnet/aspnetcore/blob/main/README.md).
```
I think this is going to cause problems as release branches get snapped.

sebastienros replied:
This is something we are already doing in most READMEs, and here I just want to prevent a dead link (or a bad one) when the folder moves.

sebastienros committed:
Update activate usage

Activate is not necessary to start VS, but it is to use `dotnet` like for starting the tests from the command line

sebastienros committed:
PR feedback

halter73 commented on src/Components/README.md:
````diff
@@ -29,42 +29,56 @@ The following contains a description of each sub-directory in the `Components` d
+```
 
-1. You'll need to install [Node](https://nodejs.org) on your machine.
+You may need to kill some processes holding on files that are being deleted, like closing Visual Studio and other `msbuild` or `dotnet` processes. The following command may help you but be aware that this could also stop other important tasks:
````
```suggestion
You may need to kill some processes holding on files that are being deleted, like closing Visual Studio and other `msbuild` or `dotnet` processes. There may also be lingering headless `chrome` processes, but they are not included in this command. The following command may help you but be aware that this could also stop other important tasks:
```

sebastienros committed:
Update src/Components/README.md

Co-authored-by: Stephen Halter <halter73@gmail.com>
